src/options/set.rs

```diff
@@ -88,7 +89,7 @@ pub fn set_options(
 
     // --color-only is used for interactive.diffFilter (git add -p) and side-by-side cannot be used
     // there (does not emit lines in 1-1 correspondence with raw git output). See #274.
-    if config::user_supplied_option("color-only", arg_matches) {
+    if config::user_supplied_option("color_only", arg_matches) {
```

Comment:
How sure are we that we've updated all places to snake-case that need updating?

Reply:
All usage of the `ArgMatches` seems to go through the `options::set::set_options` function, the `set_options!` macro and `config::user_supplied_option` function.
And the macro checks the names against the `expected_option_name_map` hash map.